criu/arch/mips/include/asm/restorer.h

```diff
@@ -0,0 +1,60 @@
+#ifndef __CR_ASM_RESTORER_H__
+#define __CR_ASM_RESTORER_H__
+
+#include "asm/types.h"
+#include <compel/asm/fpu.h>
+#include "images/core.pb-c.h"
+#include <compel/plugins/std/syscall-codes.h>
+#include <compel/asm/sigframe.h>
+
+static inline void restore_tls(tls_t *ptls) { 
+	asm volatile(							
+		     "move $4, %0				    \n"	
+		     "li $2,  "__stringify(__NR_set_thread_area)"  \n" 
+		     "syscall					    \n"	
+		     :							
+		     : "r"(*ptls)					
+		     : "$4","$2","memory");
+}
+static inline int arch_compat_rt_sigaction(void *stack, int sig, void *act)
+{
+	return -1;
+}
+static inline int set_compat_robust_list(uint32_t head_ptr, uint32_t len)
+{
+	return -1;
+}
+
+#define RUN_CLONE_RESTORE_FN(ret, clone_flags, new_sp, parent_tid,      \
+			     thread_args, clone_restore_fn)      \
+	asm volatile(							\
+		     "move $5, %6				\n"	\
+		     "move $6, %5				\n"	\
+		     "move $7, %1				\n"	\
+		     "move $8, %3				\n"	\
+		     "move $9, %4				\n"	\
+		     "break  					\n"	\
+		     : "=r"(ret)					\
```

Comment:
Having just worked on this for all other supported architectures, this looks wrong. You need to call the clone() syscall to create a new process. This just seems to move registers around without doing any syscall.

Reply:
thanks ,I will support multi-thread for mips architecture,then I will check it.

Reply:
I have support multi-thread for mips architecture